nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/trash/TrashService.java

```diff
@@ -33,20 +34,59 @@
  */
 public interface TrashService {
 
+    /***
+     * Event for a document trashed by the user. Triggers an async listener that trashes its children too.
+     *
+     * @since 10.1
+     */
+    String DOCUMENT_TRASHED = "documentTrashed";
+
+    /***
+     * Event for a document untrashed by the user. Triggers an async listener that untrashes its children too.
+     *
+     * @since 10.1
+     */
+    String DOCUMENT_UNTRASHED = "documentUntrashed";
+
+    /**
+     * Skip the renaming.
```

Comment:
Mention that this is a key for the context data, and that it happens if the value is `Boolean.TRUE`